include/dump_offload.hpp

```diff
@@ -0,0 +1,447 @@
+            (handler->second->dumpType == "resource"))
+        {
+            handler->first->close();
+            BMCWEB_LOG_CRITICAL("INFO: {} dump resetHandlers cleanup",
+                                handler->second->dumpType);
+        }
+    }
+}
+
+inline void requestRoutes(App& app)
+{
+    BMCWEB_ROUTE(
+        app,
+        "/redfish/v1/Systems/system/LogServices/Dump/Entries/<str>/attachment/")
+        .privileges({{"ConfigureComponents", "ConfigureManager"}})
+        .streamingResponse()
+        .onopen([](crow::streaming_response::Connection& conn) {
+        if (!systemHandlers.empty())
+        {
+            BMCWEB_LOG_WARNING("Can't allow dump offload opertaion, one "
+                               "of the host dump is already offloading");
+            conn.sendStreamErrorStatus(
+                boost::beast::http::status::service_unavailable);
+            conn.close();
+            return;
+        }
+
+        std::string url(conn.req.target());
+        BMCWEB_LOG_CRITICAL("INFO:Request open for system dump offload with"
+                            "url: {}",
+                            url);
+        std::string startDelimiter = "Entries/";
+        std::size_t pos1 = url.rfind(startDelimiter);
+        std::size_t pos2 = url.rfind("/attachment");
+        if (pos1 == std::string::npos || pos2 == std::string::npos)
+        {
+            BMCWEB_LOG_WARNING("Unable to extract the dump id");
+            conn.sendStreamErrorStatus(boost::beast::http::status::not_found);
+            conn.close();
+            return;
+        }
+        std::string dumpEntry =
+            url.substr(pos1 + startDelimiter.length(),
+                       pos2 - pos1 - startDelimiter.length());
+
+        // System and Resource dump entries are currently being
+        // listed under /Systems/system/LogServices/Dump/Entries/
+        // redfish path. To differentiate between the two, the dump
+        // entries would be listed as System_<id> and Resource_<id> for
+        // the respective dumps. Hence the dump id and type are being
+        // extracted here from the above format.
+        std::string dumpId;
+        std::string dumpType;
+        std::size_t idPos = dumpEntry.rfind('_');
+
+        if (idPos != std::string::npos)
+        {
+            dumpType =
+                boost::algorithm::to_lower_copy(dumpEntry.substr(0, idPos));
+            dumpId = dumpEntry.substr(idPos + 1);
+        }
+
+        boost::asio::io_context* ioCon = conn.getIoContext();
+
+        // Generating random id to create unique socket file
+        // for each dump offload request
+        std::random_device rd;
+        std::default_random_engine gen(rd());
+        std::uniform_int_distribution<> dist{0, 1024};
+        std::string unixSocketPath = std::string(unixSocketPathDir) + dumpType +
+                                     "_dump_" + std::to_string(dist(gen));
+        systemHandlers[&conn] =
```

Comment:
Why can't we just do 

```
res.openFile(unixSocketPath)
```

removes the need for any sort of monitoring and any sort of streaming registration
bmcweb will just read 4k bytes at a time from the pipe until it gets EOF or the client disconnects

Follow what @abhilashraju did at https://gerrit.openbmc.org/c/openbmc/bmcweb/+/67667/40/http/http_response.hpp

Reply:
Hi Gunnar

Reworking completely on downstream only code for every rebase is really taking more efforts than expected. especially this Dump offload streaming code is complex one.
Since this is downstream only and working code since releases, lets go ahead with this existing code
we will be moving away from this code in Pfuture releases, so I don't see much worth of reworking on this code

Reply:
more over handling file vs unix socket is different, he was doing it becoz of file_body support.

Reply:
> Reworking completely on **downstream only** code for every rebase is really taking more efforts than expected.
I second this.

Reply:
If this doesn't go away for P-Future we should rework this to be more like https://gerrit.openbmc.org/c/openbmc/bmcweb/+/67667/40/http/http_response.hpp 
This is a lot of code and I don't think most of it is needed